### DISCUSSION TITLE
Release v2.5.2

### DISCUSSION THREAD

**Discussion by pajlada** [Chatterino/chatterino2#5609] [Q&A] [answered]
Are there any bugs you've noticed between 2.5.1 and latest nightly that should be fixed before making a new release?
Are there any almost implemented features/issues that you would like implemented before the next release?
Are there any new hotkeys/default settings you think should be changed?
Let me know!

Current "pretty changelog": https://github.com/Chatterino/website/pull/617

Sorry for the ghost ping :ghost:

**Comment by pajlada** [comment]
:white_check_mark: 
I would like the transparent overlay feature in the next release
 - https://github.com/Chatterino/chatterino2/pull/4746

It adds the default Ctrl+Alt+N key binding to create an overlay window of the selected split
It adds the default Ctrl+Shift+U key binding to toggle the interactability/clickthroughness of all overlay windows

**Comment by pajlada** [comment]
:white_check_mark: 
I would like initial shared chat support in the next release:
 - https://github.com/Chatterino/chatterino2/pull/5606

**Comment by pajlada** [comment]
:white_check_mark: 
I would like the autocompletion bugs to be resolved
 - https://github.com/Chatterino/chatterino2/pull/5603
 - https://github.com/Chatterino/chatterino2/issues/5399

**Comment by jupjohn** [comment]
:white_check_mark: 
I'd personally like to see #5589 fixed as I believe it's a regression from some cleanup done post-2.5.1 (haven't confirmed, just noticed it after then). @pajlada if you've got some time to read my comment in there and give me some direction, I'm happy to fix it. 

 - #5589 :white_check_mark:

**Reply by iProdigy** [reply]
i think that bug's pretty old, but the described fix is pretty simple

**Reply by pajlada** [reply]
iProdigy's recent comment on the issue should be sufficient to fix this - feel free to hop on a fix
Thanks @jupjohn !

**Comment by Nerixyz** [comment]
This release feels like a "patch" release (we don't use semantic versioning), as there aren't any new major-major features. Looking at the changelog, we have plugins in alpha [these will be changed a bit (probably) after the release], pronouns [disabled by default], high-dpi support on Windows [Windows only]. If #4746 gets in, that would be another major feature, but I only see it as a preview (unless some more features like independent scaling are added). Even though it's a "minor" entry, removing IRC support might require a 2.6 release.
Other than that, we adapted some new Twitch features and generally improved some usability aspects.

Furthermore, this release could/should be the last to support Qt 5 (important bug fixes can still be backported). It has been [EOL since May 2023](https://www.qt.io/blog/qt-5.15-extended-support-for-subscription-license-holders). The extended (paid) subscriber LTS ends in May 2025. Other projects like [OBS](https://github.com/obsproject/obs-studio/discussions/9055) also dropped Qt 5 support recently.

It would be nice if we could make a more user-friendly changelog this time. Maybe in a blog-style like how Rust does its release-posts where they highlight a few new features and fixes.

**Reply by pajlada** [reply]
:white_check_mark: I agree v2.5.2 should be the last release to support Qt5. I made a pinned issue to let people chime in if it's a problem:
 - https://github.com/Chatterino/chatterino2/issues/5688

And I think we can try more user-friendly changelogs for the website & the GitHub release

**Comment by 8thony** [comment]
I would like to see this regression fixed:
 - #5450 :white_check_mark:

If simple enough, it would be great to fix/add:
 - #2235
 - #2673 
 -  #5565

**Reply by Nerixyz** [reply]
> I would like to see the regression #5450 fixed

I can give that a try (iirc I couldn't reproduce that the last time I tried).

> If simple enough, it would be great to fix/add #2235 

Probably not, but who knows (if someone wants to implement that, feel free to open a PR).

> #2673

That's what I wanted to work on after the message-builder refactor I mentioned in https://github.com/Chatterino/chatterino2/pull/5598.

> and #5565

Now that we have the `user:read:follows` scope, this should be simple.

**Comment by Mm2PL** [comment]
I would love to be able to finish the Lua Sol2 migration before a major release. But it's still a work-in-progress.

> Even though it's a "minor" entry, removing IRC support might require a 2.6 release.

I second this. We shouldn't remove features in patch releases. I have no idea how this ended up as a "minor".

**Reply by pajlada** [reply]
mb, it was originally (2.5.2 or 2.6.0) but I read everything in nerixyz's post except the IRC part. I'm happy with either versioning

**Reply by Felanbird** [reply]
> I have no idea how this ended up as a "minor".

always an experimental feature and while it being a major feature in theory makes sense, seeing it as minor is reasonable since like nobody uses it

**Comment by digitizeddave** [comment]
:white_check_mark: 
Is the emote picker channel order a bug?  It's started being in some kind of random order for me. The bit emotes aren't by the other channel emotes for me as well. Which are two changes that have happened from previous behavior, for me.

The behavior I used to have was current channel then alphabetical including bit emotes in their channel.   The latest nightly behavior for me is current channel -> then some random looking order with bit emotes scattered far away from the other emotes for a channel. 

But maybe this is intentional? I don't know. I can't see where this is mentioned, so I thought I might mention it. I tried looking around for this, but I am fairly stupid.

**Reply by Nerixyz** [reply]
We're now using Twitch's official API to get emotes (https://github.com/Chatterino/chatterino2/pull/5239). It has some different grouping. Please make an issue about this. I'd need to know more about what's different (since I don't have that many (bit-)emotes). Best thing would be "this is how the order/grouping was before" and "this is how it's now".

**Reply by digitizeddave** [reply]
I will try to make an issue for this right now.

**Comment by pajlada** [comment]
:white_check_mark: 
Improve `/debug-env` to print proxy information
 - https://github.com/Chatterino/Chatterino2/issues/5640

**Comment by Felanbird** [comment]
add the ubuntu 24.04 build to the `/linux` page of https://github.com/chatterino/website since this will be the first version bump with it

**Comment by pajlada** [comment]
:white_check_mark: 
https://github.com/Chatterino/chatterino2/pull/5649

**Comment by Felanbird** [comment, marked as answer]
everything looks to have been done and 2.5.2 is out
